pkg/controller/deployment/util/deployment_util.go

```diff
@@ -58,6 +58,8 @@ const (
 	RollbackTemplateUnchanged = "DeploymentRollbackTemplateUnchanged"
 	// RollbackDone is the done rollback event reason
 	RollbackDone = "DeploymentRollback"
+	// OverlapAnnotation marks deployments with overlapping selector with other deployments
+	OverlapAnnotation = "deployment.kubernetes.io/error-selector-overlapping-with"
```

Comment:
This feels more like a Deployment Condition for a permanently failed deployment ... apart from the fact that we still don't have perm-failed deployments:)